msticpy/common/azure_auth_core.py

```diff
@@ -205,16 +205,30 @@ def _az_connect_core(
     """
     # Create the auth methods with the specified cloud region
     cloud = cloud or kwargs.pop("region", AzureCloudConfig().cloud)
+    az_config = AzureCloudConfig(cloud)
+    aad_uri = az_config.endpoints.active_directory
     tenant_id = tenant_id or AzureCloudConfig().tenant_id
-    auth_options = _create_auth_options(cloud, tenant_id)
-    if not auth_methods:
-        auth_methods = default_auth_methods()
-    try:
-        auths = [auth_options[meth] for meth in auth_methods]
-    except KeyError as err:
-        raise MsticpyAzureConnectionError(
-            "Unknown authentication option, valid options are; env, cli, msi, interactive"
-        ) from err
+    if auth_methods:
+        for method in auth_methods:
+            if method in _AUTH_MAPPINGS:
+                _AUTH_MAPPINGS[method] = False
+        creds = DefaultAzureCredential(
+            authority=aad_uri,
+            exclude_cli_credential=_AUTH_MAPPINGS["cli"],
```

Comment:
Is this logic the wrong way around?
Ah no I guess not. Maybe we should rename _AUTH_MAPPINGS to something like _EXCLUDED_AUTH?
Then at least setting something in there to True makes a bit more sense.